libevm/legacy/stubs_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2025 the libevm authors.
+//
+// The libevm additions to go-ethereum are free software: you can redistribute
+// them and/or modify them under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation, either version 3 of the License,
+// or (at your option) any later version.
+//
+// The libevm additions are distributed in the hope that they will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser
+// General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see
+// <http://www.gnu.org/licenses/>.
+
+// Package legacy provides converters between legacy types and their refactored
+// equivalents.
+
+package legacy
+
+import (
+	"math/big"
+
+	"github.com/holiman/uint256"
+
+	"github.com/ava-labs/libevm/common"
+	"github.com/ava-labs/libevm/core/types"
+	"github.com/ava-labs/libevm/core/vm"
+	"github.com/ava-labs/libevm/libevm"
+	"github.com/ava-labs/libevm/params"
+)
+
+var _ vm.PrecompileEnvironment = (*stubPrecompileEnvironment)(nil)
+
+// stubPrecompileEnvironment implements [vm.PrecompileEnvironment] for testing.
+type stubPrecompileEnvironment struct {
+	gasToReturn uint64
+	gasUsed     uint64
+}
+
+// Gas returns the gas supplied to the precompile.
+func (s *stubPrecompileEnvironment) Gas() uint64 {
+	return s.gasToReturn
+}
+
+// UseGas records the gas used by the precompile.
+func (s *stubPrecompileEnvironment) UseGas(gas uint64) bool {
+	s.gasUsed += gas
+	return true
+}
+
+func (s *stubPrecompileEnvironment) Call(addr common.Address, input []byte, gas uint64, value *uint256.Int, _ ...vm.CallOption) (ret []byte, _ error) {
```

Comment:
The unused methods just become a maintenance burden that require updating any time `vm.PrecompileEnvironment` is modified. Please, instead, embed the interface in the stub. Only `Gas()` and `UseGas()` are necessary.